### PR TITLE
Update testing to cover all mjml components and important prop logic

### PR DESCRIPTION
**Pull request by emmclaughlin**
Add two test case files:

mjml-components
- Requires a test case for every component in src/mjml to ensure that every component is tested at least once
- All existing test cases from mjml-tags has been moved to this file
- These tests are set up in a way that typescript will throw an error if there is not a test for every component in src/mjml

mjml-props
- test specific prop logic. The component used for testing is arbitrary, since many of the props are the same across many components

One change was also made while testing. I noticed MjmlStyle automatically passed its children to dangerouslySetInnerHtml under the hood. This is bad because it is not obvious to the developer. Given that we have added support for the dangerouslySetInnerHtml prop, I updated this component to use children as expected.

### CODE REVIEW

File: test/__mockData__/mockMjmlReactTestData.tsx [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+    },
+  ],
+  MjmlStyle: [
+    {
+      mjmlReact: <MjmlStyle>{".button{}"}</MjmlStyle>,
+      expectedMjml: `<mj-style>.button{}</mj-style>`,
+    },
+    {
+      mjmlReact: <MjmlStyle>{"body > div {}"}</MjmlStyle>,
+      expectedMjml: `<mj-style>body &gt; div {}</mj-style>`,
```
Comment (IanEdington) [inline test/__mockData__/mockMjmlReactTestData.tsx:267]: `&gt;` I think this is why we used the dangerouslySetInnerHtml for MjmlStyle. Probably not the best to silently do a dangerous operation but not sure how to handle this case.

I don't like the idea or requiring everyone to use `dangerouslySetInnerHTML` to use styles

**Review by hchiufaire — APPROVED** [review]
looks comprehensive!

File: test/extensions.test.tsx [hunk 2]
```diff
@@ -36,8 +35,7 @@ describe("extensions", () => {
           First, solve the problem. Then, write the code.
         </MjmlConditionalComment>
       );
-      const markup = renderToMjml(comment);
-      expect(markup).toBe(
+      expect(renderToMjml(comment)).toBe(
```
Comment (IanEdington) [inline test/extensions.test.tsx:38]: I know it's a little outside the scope of this library, but I'm wondering if we should `renderToHtml`.
The downside is that tests will sometimes break on an MJML upgrade.
The upside is that we see what a test's output really would be.
We would have to figure out the skeleton.

Probably not worth it at the moment, but figured I would make the suggestion

File: test/mjml-props.test.tsx [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+    expect(renderToMjml(<MjmlTable align="right" />)).toBe(
+      `<mj-table align="right"></mj-table>`
+    );
+    expect(renderToMjml(<MjmlTable align="center" />)).toBe(
+      `<mj-table align="center"></mj-table>`
+    );
+
+    // Test with invalid props. Code should still render but typescript should
+    // throw an error
+    // @ts-expect-error invalid align prop for test purposes
```
Comment (IanEdington) [inline test/mjml-props.test.tsx:56]: I haven't used `@ts-expect-error` before. Have you tested that both having an accepted prop and not having this comment cause the test to fail?

File: test/mjml-props.test.tsx [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+    );
+    expect(renderToMjml(buttonPx)).toBe(
+      '<mj-button letter-spacing="1px">Px</mj-button>'
+    );
+    expect(renderToMjml(buttonPercentage)).toBe(
+      '<mj-button height="2%">Percent</mj-button>'
+    );
+    expect(renderToMjml(buttonEm)).toBe(
+      '<mj-button letter-spacing="3em">Em</mj-button>'
+    );
```
Comment (IanEdington) [inline test/mjml-props.test.tsx:80]: Is there a type failure case for the `px, percent, and ephemeral units`?
What type safety does it give us?

**Review by IanEdington — APPROVED** [review]
This is Awesome! Great tests!!!

I have lots of comments but non of them should block this PR from merging, except making sure that `@ts-expect-error` works as expected. I'm not sure we how much more we should invest in tests before launch of v3 so the comments are more of a brain dump of other things we could do in the future if we decide it's worth it.

File: test/__mockData__/mockMjmlReactTestData.tsx (hunk 1 above)
Comment (emmclaughlin) [reply test/__mockData__/mockMjmlReactTestData.tsx:267]: This is tricky. For this one either option is requiring everyone to use `dangerouslySetInnerHtml`, the one option just hides it under the hood.  I feel given `dangerouslySetInnerHtml` has the potential to be a security risk it is better to be super explicit that it is being used. But am open for more conversation around this

File: test/extensions.test.tsx (hunk 2 above)
Comment (emmclaughlin) [reply test/extensions.test.tsx:38]: I thought about this as well. Definitely something we should discuss further

File: test/mjml-props.test.tsx (hunk 4 above)
Comment (emmclaughlin) [reply test/mjml-props.test.tsx:80]: We have type safety via the PR @hchiufaire added https://github.com/Faire/mjml-react/pull/38

So a typo like "3en" should cause a type error.

**Pull request merged**